components/eamxx/src/physics/shoc/tests/shoc_diag_second_moments_tests.cpp

```diff
@@ -254,69 +249,70 @@ struct UnitWrap::UnitTest<D>::TestDiagSecondMoments {
     };
 
     // Assume all data is in C layout
 
-    // Get data from fortran
-    for (auto& d : f90_data) {
-      // expects data in C layout
-      diag_second_moments(d);
+    // Read baseline data
+    if (this->m_baseline_action == COMPARE) {
+      for (auto& d : baseline_data) {
+        d.read(Base::m_fid);
+      }
     }
 
     // Get data from cxx
     for (auto& d : cxx_data) {
-      d.transpose<ekat::TransposeDirection::c2f>(); // _f expects data in fortran layout
-      diag_second_moments_f(d.shcol, d.nlev, d.nlevi, d.thetal, d.qw, d.u_wind, d.v_wind,
-                            d.tke, d.isotropy, d.tkh, d.tk, d.dz_zi, d.zt_grid, d.zi_grid, d.shoc_mix,
-                            d.thl_sec, d.qw_sec, d.wthl_sec, d.wqw_sec, d.qwthl_sec, d.uw_sec,
-                            d.vw_sec, d.wtke_sec, d.w_sec);
-      d.transpose<ekat::TransposeDirection::f2c>(); // go back to C layout
+      diag_second_moments(d);
     }
 
     // Verify BFB results, all data should be in C layout
-    if (SCREAM_BFB_TESTING) {
-      static constexpr Int num_runs = sizeof(f90_data) / sizeof(DiagSecondMomentsData);
+    if (SCREAM_BFB_TESTING && this->m_baseline_action == COMPARE) {
+      static constexpr Int num_runs = sizeof(baseline_data) / sizeof(DiagSecondMomentsData);
       for (Int i = 0; i < num_runs; ++i) {
-        DiagSecondMomentsData& d_f90 = f90_data[i];
+        DiagSecondMomentsData& d_baseline = baseline_data[i];
         DiagSecondMomentsData& d_cxx = cxx_data[i];
-        for (Int k = 0; k < d_f90.total(d_f90.w_sec); ++k) {
-          REQUIRE(d_f90.w_sec[k] == d_cxx.w_sec[k]);
+        for (Int k = 0; k < d_baseline.total(d_baseline.w_sec); ++k) {
+          REQUIRE(d_baseline.w_sec[k] == d_cxx.w_sec[k]);
         }
-        for (Int k = 0; k < d_f90.total(d_f90.thl_sec); ++k) {
-          REQUIRE(d_f90.thl_sec[k] == d_cxx.thl_sec[k]);
-          REQUIRE(d_f90.qw_sec[k] == d_cxx.qw_sec[k]);
-          REQUIRE(d_f90.wthl_sec[k] == d_cxx.wthl_sec[k]);
-          REQUIRE(d_f90.wqw_sec[k] == d_cxx.wqw_sec[k]);
-          REQUIRE(d_f90.qwthl_sec[k] == d_cxx.qwthl_sec[k]);
-          REQUIRE(d_f90.uw_sec[k] == d_cxx.uw_sec[k]);
-          REQUIRE(d_f90.vw_sec[k] == d_cxx.vw_sec[k]);
-          REQUIRE(d_f90.wtke_sec[k] == d_cxx.wtke_sec[k]);
+        for (Int k = 0; k < d_baseline.total(d_baseline.thl_sec); ++k) {
+          REQUIRE(d_baseline.thl_sec[k] == d_cxx.thl_sec[k]);
+          REQUIRE(d_baseline.qw_sec[k] == d_cxx.qw_sec[k]);
+          REQUIRE(d_baseline.wthl_sec[k] == d_cxx.wthl_sec[k]);
+          REQUIRE(d_baseline.wqw_sec[k] == d_cxx.wqw_sec[k]);
+          REQUIRE(d_baseline.qwthl_sec[k] == d_cxx.qwthl_sec[k]);
+          REQUIRE(d_baseline.uw_sec[k] == d_cxx.uw_sec[k]);
+          REQUIRE(d_baseline.vw_sec[k] == d_cxx.vw_sec[k]);
+          REQUIRE(d_baseline.wtke_sec[k] == d_cxx.wtke_sec[k]);
         }
       }
+    } // SCREAM_BFB_TESTING
+    else if (this->m_baseline_action == GENERATE) {
```

Comment:
I comment here, though it's a more general question.

Why do we want to give the possibility of not generating and not comparing? What's the use case for the `-n` flag of the tests?